src/mscorlib/src/System/Reflection/Assembly.CoreCLR.cs

```diff
@@ -15,11 +15,76 @@ namespace System.Reflection
 {
     public abstract partial class Assembly : ICustomAttributeProvider, ISerializable
     {
+        private static volatile bool s_LoadFromResolveHandlerSetup = false;
+        private static object s_syncRootLoadFrom = new object();
+        private static List<string> s_LoadFromAssemblyList = new List<string>();
+        private static object s_syncLoadFromAssemblyList = new object();
+
+        private static Assembly LoadFromResolveHandler(object sender, ResolveEventArgs args)
+        {
+            Assembly requestingAssembly = args.RequestingAssembly;
+            Assembly requestedAssembly = null;
+
+            // Requesting assembly for LoadFrom is always loaded in defaultContext - proceed only if that
+            // is the case.
+            if (AssemblyLoadContext.Default == AssemblyLoadContext.GetLoadContext(requestingAssembly))
```

Comment:
Nit: I think early outs are better for this kind of code (http://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code has a good explanation why).

```
   if (AssemblyLoadContext.Default != AssemblyLoadContext.GetLoadContext(requestingAssembly))
       return null;

   string requestorPath = Path.GetFullPath(requestingAssembly.Location);
   if (string.IsNullOrEmpty(requestorPath))
       return null;
...

   if (!fRequestorLoadedViaLoadFrom)
       return null;
...
```